### PR TITLE
fix(k8stagger)!: change default pod id attribute name to k8s.pod.uid

### LINKED ISSUE #1379: [k8sprocessor] including `k8s.pod.uid` in `extract.metadata` creates `k8s.pod.id` field, but it should create `k8s.pod.uid`
Version: `v0.89.0-sumo-2`

https://github.com/SumoLogic/sumologic-otel-collector/blob/v0.89.0-sumo-2/pkg/processor/k8sprocessor/kube/kube.go#L43

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
Fixed #1379 
The previous name was k8s.pod.id.

### CODE REVIEW

**Review by sumo-drosiek — COMMENTED** [review]
We do need a migration for customers who would like to keep old behavior

**Comment by aboguszewski-sumo** [comment]
> We do need a migration for customers who would like to keep old behavior
A migration guide or a migration script/config option?

**Comment by sumo-drosiek** [comment]
> > We do need a migration for customers who would like to keep old behavior
> 
> A migration guide or a migration script/config option?

Migration guide should be enough. Not sure if we need a migration process (deprecation and so on) for this particular change. cc: @astencel-sumo

**Comment by astencel-sumo** [comment]
I think an upgrade doc in the Upgrading.md file should be fine, explaining what's changed and how to revert to the previous behavior if the user needs that.

File: docs/upgrading.md [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Upgrading
 
 - [Upgrading](#upgrading)
-  - [Upgrading to v0.90.1-sumo-0](#upgrading-to-v0910-sumo-0)
+  - [Upgrading to v0.91.0-sumo-0](#upgrading-to-v0910-sumo-0)
```
Comment (sumo-drosiek) [inline docs/upgrading.md:4]: missing sub point

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**